### ISSUE TITLE
CS8604 False-negative in .Net Standard project vs .Net Core

### ISSUE THREAD

**Issue by Pyrobolser** [dotnet/roslyn-analyzers#3993]
### Describe the bug

In a .Net Standard project, trying to use a `Stream?` instead of a `Stream` raises a CS8604 warning. But not when using a `Stream?` to initialize a `StreamReader`.
In a .Net Standard project, both situations raise a CS8604 warning.

### Steps To Reproduce
Here is the short class that I am using, with `<Nullable>enable</Nullable>` in .NetStandard 2.1 project and .NetCoreApp 3.1 project.

```cs
public class Test
{
    public async Task<Stream?> GetFile(string file)
    {
        Stream? stream = null;

        if (File.Exists(file))
        {
            stream = File.OpenRead(file);
        }

        return await Task.FromResult(stream);
    }

    public async Task TestStreamReader(string file)
    {
        using(Stream stream = await GetFile(file))
        {
            // Do stuff...
        }

        using(StreamReader reader = new StreamReader(await GetFile(file)))
        {
            // Do stuff...
        }
    }
}
```

### Expected behavior

CS8604 for both using `Stream` and `StreamReader` in .Net Standard and .Net Core

### Actual behavior

CS8604 for `Stream` in .Net Standard and for `Stream` and `StreamReader` in .Net Core

### Analyzer

**Package**: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

**Version**: v3.3.0 (Latest)

**Diagnostic ID**: [CA1716](https://docs.microsoft.com/visualstudio/code-quality/ca1716)

## Additional context

I understand this might be "intentional" as it should only work properly in .Net Core App if I understand this correctly but I feel like the fact that this is "half" working in .Net Standard can make it confusing.
